### DISCUSSION TITLE
Mesh won't run after 1.22 update - How to restore to older version

### DISCUSSION THREAD

**Discussion by ghost** [Ylianst/MeshCentral#5954] [Issues]
Hi,
It's already a known bug so I just wanted to ask how can I restore back to older version?
I can't run the Mesh as I'm getting a Permission denied error: and the service is configured as needed.
When I look in the Errors file I see:
```
/home/quizhead/meshcentral/node_modules/meshcentral/node_modules/minimatch/dist/cjs/ast.js:86
    #fillNegs() {
             ^

SyntaxError: Unexpected token '('
    at wrapSafe (internal/modules/cjs/loader.js:915:16)
    at Module._compile (internal/modules/cjs/loader.js:963:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/quizhead/meshcentral/node_modules/meshcentral/node_modules/minimatch/dist/cjs/index.js:9:18)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
```
Should I delete the Node Modules folder and reinstall?

**Comment by si458** [comment]
what node version are you using?
after looking at the other bug report, you need to use node 18 or above.

**Reply by ghost** [reply]
According to node -v command I'm using v20.11.1

**Reply by si458** [reply]
ok so the `Permission denied error` is basiclly you need to check all the files and how you run node are the same user
1. `ls -lah /FOLDERTOMESHCENTRAL/meshcentral-data/` - check the user/group in 3rd/4th columns for the `.db` files
2. make sure the user you are logged in as is the same as the `.db` files are
3. again check `cat /etc/systemd/system/meshcentral.service` that user/group MATCHES the `.db` files
4. if `.db` files DONT match then you can fix this by running `sudo chown -R USER:GROUP /FOLDERTOMESHCENTRAL/meshcentral-data/`

**Reply by ghost** [reply]
Ok I checked and everything is configured as needed,
The first time I ran sudo chown -R USER:GROUP /FOLDERTOMESHCENTRAL/meshcentral-data/ in worked localy and after PC restart again not working

**Reply by si458** [reply]
so the must be an issue with again file permissions OR node versions,
can you share the outputs from `ls -lah /FOLDERTOMESHCENTRAL/meshcentral-data/` and `cat /etc/systemd/system/meshcentral.service`

**Reply by ghost** [reply]
quizhead@Latitude-5480:~/meshcentral/node_modules$ ls -lah /home/quizhead/meshcentral/meshcentral-data/
total 12M
drwxrwxrwx 3 quizhead quizhead 4.0K Mar 25 16:11 .
drwxrwxrwx 7 quizhead quizhead 4.0K Mar 15 19:50 ..
-rw-rw-rw- 1 quizhead quizhead 2.5K Feb 14 11:29 agentserver-cert-private.key
-rw-rw-rw- 1 quizhead quizhead 1.6K Feb 14 11:29 agentserver-cert-public.crt
-rw-rw-rw- 1 quizhead quizhead 2.5K Feb 14 11:29 codesign-cert-private.key
-rw-rw-rw- 1 quizhead quizhead 1.6K Feb 14 11:29 codesign-cert-public.crt
-rw-rw-rw- 1 quizhead quizhead  764 Mar 22 15:12 config.json
-rw-r--r-- 1 quizhead quizhead  22K Feb 15 18:20 logoback.png
-rw-r--r-- 1 root     root     183K Mar 25 16:11 meshcentral.db
-rw-r--r-- 1 root     root     4.4M Mar 25 16:11 meshcentral-events.db
-rw-r--r-- 1 root     root     134K Mar 25 16:11 meshcentral-power.db
-rw-r--r-- 1 root     root     1.3M Mar 25 16:11 meshcentral-stats.db
-rw-r--r-- 1 quizhead quizhead 1.1M Mar 25 16:11 mesherrors.txt
-rw-rw-rw- 1 quizhead quizhead 1.7K Mar  5 20:20 mpsserver-cert-private.key
-rw-rw-rw- 1 quizhead quizhead 1.7K Feb 14 11:29 mpsserver-cert-private.key-old
-rw-rw-rw- 1 quizhead quizhead 1.7K Feb 14 18:13 mpsserver-cert-private.key-old2
-rw-rw-rw- 1 quizhead quizhead 1.7K Mar  5 20:18 mpsserver-cert-private.key-old3
-rw-rw-rw- 1 quizhead quizhead 1.5K Mar  5 20:20 mpsserver-cert-public.crt
-rw-rw-rw- 1 quizhead quizhead 1.5K Feb 14 11:29 mpsserver-cert-public.crt-old
-rw-rw-rw- 1 quizhead quizhead 1.5K Feb 14 18:13 mpsserver-cert-public.crt-old2
-rw-rw-rw- 1 quizhead quizhead 1.5K Mar  5 20:18 mpsserver-cert-public.crt-old3
-rw-rw-rw- 1 quizhead quizhead 2.5K Feb 14 11:29 root-cert-private.key
-rw-rw-rw- 1 quizhead quizhead 1.6K Feb 14 11:29 root-cert-public.crt
-rw-rw-rw- 1 quizhead quizhead   94 Mar 25 16:11 serverstate.txt
drwxrwxrwx 2 quizhead quizhead 4.0K Feb 14 11:29 signedagents
-rw-rw-rw- 1 quizhead quizhead 2.5K Mar  5 20:20 webserver-cert-private.key
-rw-rw-rw- 1 quizhead quizhead 2.5K Feb 14 11:29 webserver-cert-private.key-old
-rw-rw-rw- 1 quizhead quizhead 2.5K Feb 14 18:13 webserver-cert-private.key-old2
-rw-rw-rw- 1 quizhead quizhead 2.5K Mar  5 20:18 webserver-cert-private.key-old3
-rw-rw-rw- 1 quizhead quizhead 1.7K Mar  5 20:20 webserver-cert-public.crt
-rw-rw-rw- 1 quizhead quizhead 1.7K Feb 14 11:29 webserver-cert-public.crt-old
-rw-rw-rw- 1 quizhead quizhead 1.7K Feb 14 18:13 webserver-cert-public.crt-old2
-rw-rw-rw- 1 quizhead quizhead 1.7K Mar  5 20:18 webserver-cert-public.crt-old3
-rw-r--r-- 1 quizhead quizhead 4.7M Mar 14 18:52 WelcomePic.JPG

quizhead@Latitude-5480:~/meshcentral/node_modules$ cat /etc/systemd/system/meshcentral.service
[Unit]
Description=MeshCentral Server
After=network.target

[Service]
Type=simple
User=root
Group=root
WorkingDirectory=/home/quizhead/meshcentral/node_modules/meshcentral
ExecStart=/usr/bin/node meshcentral

[Install]
WantedBy=multi-user.target

**Reply by si458** [reply]
you are mixing user/groups again
your databases are all `root:root` where as your certificates are `quizhead:quizhead`
the whole folder `meshcentral-data` and files inside it needs to be the same user/group!
so try this:
1. `sudo chown quizhead:quizhead  -R /home/quizhead/meshcentral/meshcentral-data/`
2. change the `meshcentral.service` to `quizhead` for user/group
3. `sudo setcap 'cap_net_bind_service=+ep' /usr/bin/node`
4. restart server

**Reply by ghost** [reply]
I didn't mix it as the db. files user are the same as the user in the service file and it worked like this for some time because without it I was getting the permission error
When I ran it manually the db files return to be root although I configured as said above.
Restarting after the above did help.
Downgrading to 1.0 version made it work again with root user in the db files and the service file.
